From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Fri, 27 May 2022 14:06:59 -0700
Subject: [tsflower] Resolve re-export indirection on CommonActions
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

What's happening here is that this file has in the .d.ts:

    import type { /*…*/ CommonActions } from '@react-navigation/routers';

and the latter module has:

    import * as CommonActions from './CommonActions';
    export { CommonActions };

and then finally the `CommonActions.d.ts` file has:

    export declare type Action = {

So when this file goes on to make a type reference
`CommonActions.Action`, that's using the `CommonActions` namespace
which it imported from react-nav/routers; where the latter made that
whole namespace be one of its exports.

TsFlower should handle this case.  There's no direct analogue to it in
Flow -- the only namespace a type can live in is a whole module, from
a namespace import like `import * as React from 'react'`.  But we can
always rewrite the type references to single names, and emit direct
imports for those.
---
 .../core/lib/typescript/src/getActionFromState.js.flow     | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git types/@react-navigation/core/lib/typescript/src/getActionFromState.js.flow types/@react-navigation/core/lib/typescript/src/getActionFromState.js.flow
index 72b0fdb2c..2f13f9fe3 100644
--- types/@react-navigation/core/lib/typescript/src/getActionFromState.js.flow
+++ types/@react-navigation/core/lib/typescript/src/getActionFromState.js.flow
@@ -1,11 +1,8 @@
 /* @flow
  * @generated by TsFlower
  */
-import {
-  typeof CommonActions,
-  type NavigationState,
-  type PartialState,
-} from '@react-navigation/routers';
+import { type NavigationState, type PartialState } from '@react-navigation/routers';
+import * as CommonActions from '@react-navigation/routers/lib/typescript/src/CommonActions';
 import { type NavigatorScreenParams, type PathConfigMap } from './types';
 
 type Options = {
-- 
2.32.0

